atest/acceptance/windows.robot

```diff
@@ -169,7 +175,7 @@ Select Popup Window With Delay By Excluded List
     @{excluded_handle_list}=    Get Window Handles
     Click Button     id:MyButton
     Switch Window    ${excluded_handle_list}    timeout=5
-    Title Should Be    Original
+    Wait Until Keyword Succeeds    5s    200ms    Title Should Be    Original
```

Comment:
I'm a bit cautious with this approach and change. It is not surprising to me that over time, for tests that were once running fine timing issues appear. I guess what I want to ask is that is a timing issue does appear is there any deeper understanding as to why. I suspect in most cases there is not. I just don't want to be masking issues in the tests and timing issues, especially when the appear where they were once not there, is something that throws alarm bells in my head.

I would also like our tests to be a model for how others should right tests, as best as we can. I know this is an ever difficult battle to fight or steep hill to climb. I'm not a fan of Wait Until Keyword Succeeds (WUKS) but as we don't have yet implemented the [generic wait for element state](https://github.com/robotframework/SeleniumLibrary/issues/1767#issuecomment-1016653989) then this is fine.

Reply:
I think this was fine on Chrome but Firefox had timing problems, I will try to confirm this later. The window is selected correctly, just that sometimes it would read the title of the old window. My guess is that switching windows is a blocking action with chromedriver but not blocking with geckodriver.

Reply:
Just thinking this through

- This test case purposes states via its test case name "With Delay". And there is another similar test case; one without delay "Select Popup Window By Excluded List"
- We want/expect it to be successful. This can be said because of any lack of expect error or other validation code to the opposite. But is this test testing that with the delay it should be "old title" that we are expecting. Or  are we expecting a blocking action in all cases and thus the title should be the "new title".
- With the timeout argument on the Switch Window call being 5[secs] should the timeout on WUKS be around the same? or more? or less? Basically how do these two interact is anything?

Reply:
The difference between `Select Popup Window By Excluded List` and `Select Popup Window With Delay By Excluded List` is that "With Delay" clicks a button that sets a timer for 1000 milliseconds that then opens the popup, while the first test clicks a link that opens it immediately.

Both tests first navigate to [popupwindow.html](https://github.com/Brownies/SeleniumLibrary/blob/fix-tests-on-firefox/atest/resources/html/javascript/popupwindow.html) and the title of that page is `Click link to show a popup window`. The popup that is opened is [dynamic_content.html](https://github.com/Brownies/SeleniumLibrary/blob/fix-tests-on-firefox/atest/resources/html/javascript/dynamic_content.html) that, somewhat confusingly for these tests, has title `Original`.

The WUKS timeout probably doesn't matter too much as long as it retries at least a couple of times. I don't think it has any interaction directly with `Switch Window`.

Reply:
Thus the idea that Switch Window wouldn't return to the title had adjusted. Ok, let's leave the WUKS in there.